fontbe/src/features.rs

```diff
@@ -180,19 +268,27 @@ impl<'a> FeatureWriter<'a> {
 
             match (left, right) {
                 (KernParticipant::Glyph(left), KernParticipant::Glyph(right)) => {
-                    let gid0 = name_to_gid(left);
-                    let gid1 = name_to_gid(right);
+                    let gid0 = self.glyph_id(left)?;
+                    let gid1 = self.glyph_id(right)?;
                     ppos_subtables.insert_pair(gid0, x_adv_record, gid1, empty_record);
                 }
                 (KernParticipant::Group(left), KernParticipant::Group(right)) => {
-                    let left = glyph_classes.get(left).unwrap().clone();
-                    let right = glyph_classes.get(right).unwrap().clone();
+                    let left = glyph_classes
+                        .get(left)
+                        .ok_or_else(|| Error::MissingGlyphId(left.clone()))?
```

Comment:
I wonder if we could pre-validate the kerning info at an earlier point in the pipeline?